packages/react-components/src/components/Datepicker/RangeDatePicker.tsx

```diff
@@ -0,0 +1,248 @@
+    }
+
+    const optionsHash = props.options.reduce(
+      (
+        acc: { [key: string]: IRangeDatePickerOption },
+        option: IRangeDatePickerOption
+      ) => ({ ...acc, [option.id]: option }),
+      {}
+    );
+
+    props.onChange(optionsHash[selectedItem]);
+  }, [props.onChange, state.selectedItem, props.options]);
+
+  const handleDayMouseEnter = React.useCallback(
+    (day: Date) => {
+      const isInRange = props.toMonth
+        ? differenceInCalendarDays(props.toMonth, day) >= 0
+        : true;
+
+      if (!isSelectingFirstDay(state.from, state.to) && isInRange) {
+        dispatch({
+          type: RangeDatePickerAction.NEW_TEMPORARY_TO_VALUE,
+          payload: { date: day },
+        });
+      }
+    },
+    [props.toMonth, state.from, state.to]
+  );
+
+  const handleDayClick = React.useCallback(
+    (day: Date) => {
+      const { from, to } = state;
+
+      if (!isDateWithinRange(day, { to: props.toMonth })) {
+        return;
+      }
+
+      if (isSelectingFirstDay(from, to)) {
+        dispatch({
+          type: RangeDatePickerAction.SELECT_FIRST_DAY,
+          payload: { date: day },
+        });
+      } else if (
+        (from && isSameDay(day, from)) ||
+        (from && isAfter(day, from))
+      ) {
+        dispatch({
+          type: RangeDatePickerAction.SELECT_SECOND_DAY_AS_TO,
+          payload: { date: day },
+        });
+      } else {
+        dispatch({
+          type: RangeDatePickerAction.SELECT_SECOND_DAY_AS_FROM,
+          payload: { date: day },
+        });
+      }
+    },
+    [props.toMonth, state.from, state.to]
+  );
+
+  const handleItemSelect = React.useCallback(
+    (itemKey: string) => {
+      if (itemKey === null) {
+        dispatch({
+          type: RangeDatePickerAction.NEW_SELECTED_ITEM,
+          payload: { selectedItem: null },
+        });
+        return;
+      }
+
+      const selectedOption = getSelectedOption(itemKey, props.options);
+
+      if (!selectedOption) {
+        return;
+      }
+
+      dispatch({
+        type: RangeDatePickerAction.NEW_SELECTED_ITEM,
+        payload: { selectedItem: itemKey },
+      });
+    },
+    [props.options]
+  );
+
+  const handleMonthChange = React.useCallback((date: Date) => {
+    dispatch({
+      type: RangeDatePickerAction.CURRENT_MONTH_CHANGE,
+      payload: { date },
+    });
+  }, []);
+
+  const getRangeDatePickerApi = (): IRangeDatePickerChildrenPayload => {
+    const { currentMonth, from, selectedItem, temporaryTo, to } = state;
+    const modifiers = React.useMemo(
+      () => getRangeDatePickerModifiers(from, temporaryTo),
+      [from, temporaryTo]
+    );
+    const selectedOption = React.useMemo(() => {
+      return getSelectedOption(selectedItem, props.options);
+    }, [props.options, selectedItem]);
+
+    const selectedDays = React.useMemo(() => {
+      return [from, { from, to: temporaryTo }];
+    }, [from, temporaryTo]);
+
+    const disabledDays = React.useMemo(() => {
```

Comment:
`React.` prefix makes a lot of noise here. Consider importing useMemo and other hooks directly from React

Reply:
I am aware, but it has become kinda standard among components in DS. If I am to change that we would have to all agree that we want to make that change across all components. Personally, I don't mind that prefix @sgraczyk, @marcinsawicki ?

Reply:
Ok. Just bear in mind you could benefit from some optimizations using the new JSX transform in React 17 in the future https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html#whats-different-in-the-new-transform 

> Depending on your setup, its compiled output may slightly improve the bundle size.

Reply:
we have created a task for that https://github.com/livechat/design-system/issues/379